libraries/AP_Scheduler/PerfInfo.h

```diff
@@ -0,0 +1,34 @@
+#include <stdint.h>
+
+namespace AP {
+
+class PerfInfo {
+public:
+
+    void reset();
+    void ignore_this_loop();
+    void check_loop_time(uint32_t time_in_micros);
+    uint16_t get_num_loops() const;
+    uint32_t get_max_time() const;
+    uint32_t get_min_time() const;
+    uint16_t get_num_long_running() const;
+    uint32_t get_num_dropped() const;
+    uint32_t get_avg_time() const;
+    uint32_t get_stddev_time() const;
+
+private:
+
+    // 400hz loop update rate
+#define OVERTIME_THRESHOLD_MICROS 3000
```

Comment:
Please move to cpp file, doesn't look to be needed anywhere else.

Reply:
Done